subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/deps/ClassSetAnalysis.java

```diff
@@ -45,16 +48,20 @@
 
     private final ClassSetAnalysisData classAnalysis;
     private final AnnotationProcessingData annotationProcessingData;
+    private final CompilerApiData compilerApiData;
     private final ImmutableSetMultimap<String, String> classDependenciesFromAnnotationProcessing;
     private final ImmutableSetMultimap<String, GeneratedResource> resourceDependenciesFromAnnotationProcessing;
+    private final Map<String, Set<String>> accessibleConstantDependentsCache = new Object2ObjectOpenHashMap<>(100);
```

Comment:
:x: Have you identified this as an actual bottleneck while profiling? If not, please remove this for now. We should keep the code simple whenever possible.

Reply:
Make sense. Removed, we will probably see if there is any issue.